### PR TITLE
Don't automatically copy config files

### PR DESCRIPTION
**Pull request by cdcapano**
Currently, when you initialize a `WorkflowConfigParser` it will automatically (and silently) copy the config file to the current working directory. This is done by the call to `resolve_url` that occurs in the `__init__`. While this make sense for remote files, it can lead to unexpected files getting created if the file exists locally.

This can be problematic when doing inference-related things. `WorkflowConfigParser` is used extensively throughout the inference package for specifying priors, models, samplers, etc. If you run `pycbc_inference` and point it to a configuration file not in the CWD, you'll get a copy of the file in your CWD which is never used for anything. That can be confusing. For example, it's happened to me where I edited the CWD copy after an initial inference run, forgetting that the run script was pointing to another file. The edits where then immediately deleted when I re-ran.

`WorkflowConfigParser` is also used by `pycbc_inference_plot_(posterior|prior)` in order to plot priors, and by `pycbc_create_injections` to create a set of injections. All of these lead to spurious copies of the configuration file.

This patch fixes this by adding a `copy_to_cwd` keyword argument to `resolve_url` and `WorkflowConfigParser.__init__`. A copy of a local file will only be created if this is set to True (it is False by default). If the file is a remote file, it will still create a copy.

I've tested this with pycbc inference. I'm not sure how important this feature is for the other workflows though.

### CODE REVIEW

**Comment by spxiwh** [comment]
Thanks for the patch Collin. `resolve_url` is used a lot in the workflow code, not just in terms of config files. It's used to bring in any File input arguments for codes that need them for example.

I think that all of this would still run fine with `copy_to_cwd=False` but I'm wondering if that's the right thing to do by default. It's kind of nice to have working directories that contain all the stuff used to create any given workflow.

Another example is this one: https://github.com/gwastro/pycbc/pull/3112/files. This code *will* be broken by this patch as moving and not copying the file would not be the desired behaviour in this case (although just using `shutil.copy` would fix this).

So, in short, I'm not convinced that the default should be `False` and not `True` given the workflow use case of this function, but if the current default behaviour does change it's worth checking what other places are using this function, and how this might affect them.

**Comment by ahnitz** [comment]
@cdcapano I agree with @spxiwh. I think the current behavior for "resolve_url" should be the default in that function. However, I do think the WorkflowConfigParser would be ok to set this argument to false. There will already be a copy of the config files made in the saved parsed version, and the copied ones *do* create some confusion as they aren't actually being used in any way if the workflow is regenerated with the same command line (i.e. they'll just be overwritten again).

**Comment by spxiwh** [comment]
I agree with @ahnitz 's suggestion

**Comment by cdcapano** [comment]
@ahnitz @spxiwh Sounds good... I've made the switch; `resolve_url` defaults to copy (same as old behavior) whereas `WorkflowConfigParser` defaults to not copy.

**Comment by cdcapano** [comment]
@spxiwh or @ahnitz Can this be merged now?
